src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/LeafStrategy.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Uno.Extensions.Collections;
+using Uno.Extensions.Collections.Tracking;
+using Uno.Extensions.Reactive.Bindings.Collections._BindableCollection.Facets;
+using Uno.Extensions.Reactive.Bindings.Collections._BindableCollection.Views;
+using TrackingMode = Uno.Extensions.Collections.TrackingMode;
+
+namespace Uno.Extensions.Reactive.Bindings.Collections._BindableCollection.Data
+{
+	/// <summary>
+	/// A holder of a leaf in a tree of nested <see cref="IObservableCollection"/>, or the root collection in case on non grouped collections
+	/// </summary>
+	internal class LeafStrategy : IBindableCollectionDataLayerStrategy
+	{
+		private readonly DataStructure _dataStructure;
+		private readonly CollectionAnalyzer _diffAnalyzer;
+		private readonly bool _isRoot;
+
+		public LeafStrategy(DataStructure dataStructure, CollectionAnalyzer diffAnalyzer, bool isRoot)
+		{
+			_dataStructure = dataStructure;
+			_diffAnalyzer = diffAnalyzer;
+			_isRoot = isRoot;
+		}
+
+		public (CollectionFacet source, ICollectionView view, IEnumerable<object> facets) CreateView(IBindableCollectionViewSource source)
+		{
+			var view = default(ICollectionView);
+
+			var collectionChangedFacet = new CollectionChangedFacet(() => view ?? throw new InvalidOperationException("The owner provider must be resolved lazily!"));
```

Comment:
What does it mean, what can the user do about it ?

Reply:
The user cannot do anything: it's an internal usage error. I used delegate here only to break a circular reference, and I raise the exception to not just put a silent `!`. We are creating "facets" for a `ICollectionView` which does need those facets to work. 

For instance the `CollectionChangedFacet` is responsible to implement and raise events the `INotifyCollectionChanged` aspect of the `ICollectionView`. The `view` needs to know this "facet" to forward add/remove on the `CollectionChanged` event. And the "facet" needs to known the `view` to use it as `sender` when it raises those events.